File: backend/test_cors.js
const http = require('http');

const data = JSON.stringify({
  email: 'test-node@test.com',
  password: 'password123'
});

// Test OPTIONS
const reqOptions = http.request({
  hostname: 'localhost',
  port: 8080,
  path: '/api/v1/auth/register',
  method: 'OPTIONS',
  headers: {
    'Origin': 'http://localhost:8087',
    'Access-Control-Request-Method': 'POST'
  }
}, (res) => {
  console.log(`OPTIONS STATUS: ${res.statusCode}`);
  console.log('OPTIONS HEADERS:', res.headers);
});
reqOptions.on('error', (e) => console.error(`OPTIONS error: ${e.message}`));
reqOptions.end();

// Test POST
const reqPost = http.request({
  hostname: 'localhost',
  port: 8080,
  path: '/api/v1/auth/register',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Origin': 'http://localhost:8087',
    'Content-Length': Buffer.byteLength(data)
  }
}, (res) => {
  console.log(`POST STATUS: ${res.statusCode}`);
  console.log('POST HEADERS:', res.headers);
  res.setEncoding('utf8');
  res.on('data', (chunk) => console.log(`POST BODY: ${chunk}`));
});
reqPost.on('error', (e) => console.error(`POST error: ${e.message}`));
reqPost.write(data);
reqPost.end();

File: backend/test_cors.py
import urllib.request
import json

url = "http://localhost:8080/api/v1/auth/register"
data = json.dumps({"email": "test-py@test.com", "password": "password123"}).encode('utf-8')

# Test OPTIONS
req_options = urllib.request.Request(url, method="OPTIONS")
req_options.add_header("Origin", "http://localhost:8087")
req_options.add_header("Access-Control-Request-Method", "POST")
try:
    with urllib.request.urlopen(req_options) as res:
        print("OPTIONS Response:", res.status)
        print("OPTIONS Headers:", res.headers)
except Exception as e:
    print("OPTIONS Error:", e)

# Test POST
req_post = urllib.request.Request(url, data=data, method="POST")
req_post.add_header("Content-Type", "application/json")
req_post.add_header("Origin", "http://localhost:8087")
try:
    with urllib.request.urlopen(req_post) as res:
        print("POST Response:", res.status)
        print("POST Headers:", res.headers)
        print("POST Body:", res.read().decode('utf-8'))
except Exception as e:
    print("POST Error:", e)
